api/models/ml_model.py

```diff
@@ -0,0 +1,133 @@
+            raise ValueError('Not valid ML model. Try one of the following: "openvino", "openpifpaf_tensorrt",'
+                             '"mobilenet_ssd_v2", "openpifpaf", "yolov3",'
+                             '"ssd_mobilenet_v2_coco", "ssd_mobilenet_v2_pedestrian_softbio", "posenet",'
+                             '"pedestrian_ssd_mobilenet_v2", "pedestrian_ssdlite_mobilenet_v2".')
+        return name
+
+    @validator("image_size")
+    def validate_image_size(cls, image_size):
+        integers = image_size.split(",")
+
+        if len(integers) != 3:
+            raise ValueError('ImageSize must be a string with 3 numbers separated with commas. Ex: "30,30,40".')
+
+        for x in integers:
+            try:
+                int(x)
+            except ValueError:
+                raise ValueError('ImageSize must be a string with 3 numbers separated with commas. Ex: "30,30,40".')
+
+        return image_size
+
+    # Root validators are called after each field validators success.
+
+    @root_validator(skip_on_failure=True)
+    def check_models_and_device(cls, values):
+        if values.get("device") == "Jetson":
+            if values.get("name") not in ["ssd_mobilenet_v2_coco", "ssd_mobilenet_v2_pedestrian_softbio",
+                                          "openpifpaf_tensorrt"]:
+                raise ValueError(f'The device {values.get("device")} only supports the following models:'
+                                 f'"ssd_mobilenet_v2_coco", "ssd_mobilenet_v2_pedestrian_softbio",'
+                                 f'"openpifpaf_tensorrt". ')
+        elif values.get("device") == "EdgeTPU":
+            if values.get("name") not in ["mobilenet_ssd_v2", "pedestrian_ssd_mobilenet_v2",
+                                          "pedestrian_ssdlite_mobilenet_v2", "posenet"]:
+                raise ValueError(f'The device {values.get("device")} only supports the following models:'
+                                 f'"mobilenet_ssd_v2", "pedestrian_ssd_mobilenet_v2","pedestrian_ssdlite_mobilenet_v2",'
+                                 f'"posenet". ')
+        elif values.get("device") == "Dummy":
+            # No restrictions on this model.
+            pass
+        elif values.get("device") in ["x86", "x86-gpu"]:
+            if values.get("name") not in ["mobilenet_ssd_v2", "openvino", "openpifpaf", "openpifpaf_tensorrt",
+                                          "yolov3"]:
+                raise ValueError(f'The device {values.get("device")} only supports the following models:'
+                                 f'"mobilenet_ssd_v2", "openvino","openpifpaf","openpifpaf_tensorrt"'
+                                 f'"yolov3". ')
+
+        return values
+
+    @root_validator(skip_on_failure=True)
+    def check_variables_for_models(cls, values):
+
+        if values.get("name") == "ssd_mobilenet_v2_coco" or values.get("name") == "ssd_mobilenet_v2_pedestrian_softbio":
```

Comment:
Maybe we can simplify all the `pass` sentences in a single line:

Something like 

```
if values.get("name) in ["ssd_mobilenet_v2_coco",'"ssd_mobilenet_v2_pedestrian_softbio", ... ]:
    pass
```

Reply:
Okey, no problem.

Reply:
Done!